connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/DeadLetterQueueReporter.java

```diff
@@ -46,12 +53,26 @@
 
     private static final int DLQ_NUM_DESIRED_PARTITIONS = 1;
 
+    public static final String ERROR_HEADER_PREFIX = HEADER_PREFIX + ".errors";
+    public static final String ERROR_HEADER_ORIG_TOPIC = ERROR_HEADER_PREFIX + ".topic";
+    public static final String ERROR_HEADER_ORIG_PARTITION = ERROR_HEADER_PREFIX + ".partition";
+    public static final String ERROR_HEADER_ORIG_OFFSET = ERROR_HEADER_PREFIX + ".offset";
+    public static final String ERROR_HEADER_CONNECTOR_NAME = ERROR_HEADER_PREFIX + ".connector.name";
+    public static final String ERROR_HEADER_TASK_ID = ERROR_HEADER_PREFIX + ".task.id";
+    public static final String ERROR_HEADER_STAGE = ERROR_HEADER_PREFIX + ".stage";
+    public static final String ERROR_HEADER_EXECUTING_CLASS = ERROR_HEADER_PREFIX + ".class.name";
+    public static final String ERROR_HEADER_EXECPTION = ERROR_HEADER_PREFIX + ".exception.class.name";
+    public static final String ERROR_HEADER_EXECPTION_MESSAGE = ERROR_HEADER_PREFIX + ".exception.message";
+    public static final String ERROR_HEADER_EXECPTION_STACK_TRACE = ERROR_HEADER_PREFIX + ".exception.stacktrace";
```

Comment:
I'd move the `ERROR_HEADER_PREFIX` here since it really is more of an implementation detail. And as @kkonstantine suggested, I'd change the prefix to be `__connect.errors.` (with the trailing period) so that the above lines can be like:

    public static final String ERROR_HEADER_PREFIX = HEADER_PREFIX + "errors";